client/src/components/ObjectStore/Instances/EditInstance.vue

```diff
@@ -36,8 +37,21 @@ const loadingMessage = "Loading storage location template and instance informati
 async function onSubmit(formData: any) {
     if (template.value) {
         const payload = editFormDataToPayload(template.value, formData);
-        const args = { user_object_store_id: String(instance?.value?.uuid) };
-        const { data: objectStore } = await update({ ...args, ...payload });
+
+        const user_object_store_id = instance?.value?.uuid!;
+
+        const { data: objectStore, error } = await GalaxyApi().PUT(
+            "/api/object_store_instances/{user_object_store_id}",
+            {
+                params: { path: { user_object_store_id } },
+                body: payload,
+            }
+        );
+
+        if (error) {
+            rethrowSimple(error);
```

Comment:
This seems to be only handled at https://github.com/galaxyproject/galaxy/blob/dev/client/src/entry/analysis/router.js#L739-L743, if this component is ever mounted as a subcomponent there would be no error handling at all, I think ?
All of this to say maybe this needs some error handling inside of the component ?

Reply:
Which doesn't need to happen now, it isn't worse than before.

Reply:
I agree, in general, we need better error handling in most components. I didn't want to add many changes to the already monstrous PR, but we should definitely be able to search for those specific `rethrowSimple` calls and maybe transform them to better error handling in the components as a follow-up. At least it is easier to find those spots now :smiley: